functests/1_performance/rt-kernel.go

```diff
@@ -2,23 +2,51 @@ package __performance
 
 import (
 	"context"
-	"fmt"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
 	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
 	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/discovery"
 	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
 	"github.com/openshift-kni/performance-addon-operators/functests/utils/pods"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+	performancev1alpha1 "github.com/openshift-kni/performance-addon-operators/pkg/apis/performance/v1alpha1"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 var _ = Describe("[performance]RT Kernel", func() {
 	var testpod *corev1.Pod
 
+	testutils.BeforeAll(func() {
+		profile := &performancev1alpha1.PerformanceProfile{}
+		key := types.NamespacedName{
+			Name: testutils.PerformanceProfileName,
+		}
+		err := testclient.Client.Get(context.TODO(), key, profile)
+		Expect(err).ToNot(HaveOccurred(), "failed to get the performance profile")
+
+		profilesCount, err := profiles.GetProfilesCount()
+		Expect(err).ToNot(HaveOccurred())
+		excludedProfiles := []string{}
+
+		for profilesCount > 0 {
```

Comment:
Not sure I get the purpouse here. You check the existance of _the_ profile, and look for alternatives.
I guess you can remove the initial `Get` as we you are looking for a specific one.
Also, passing a filter function to GetDiscoveryPerformanceProfile instead of a list would make this more efficient, as you would get the list only once.